protos/table.proto

```diff
@@ -369,4 +369,17 @@ message BitmapIndexDetails {}
 message LabelListIndexDetails {}
 message InvertedIndexDetails {}
 message NGramIndexDetails {}
-message VectorIndexDetails {}
\ No newline at end of file
+message VectorIndexDetails {}
+
+message FragmentReuseIndexDetails {
+
+  message Version {
+    // version of the dataset corresponds to the dataset_version at the time the index adds this version entry
+    uint64 dataset_version = 1;
```

Comment:
If we want to use this for conflict resolution, we should probably add this index in the same commit as compaction.

Reply:
Yes that is the goal, this should be the same dataset version as the compaction